docs/source/model/dependencies.rst

```diff
@@ -394,6 +394,100 @@ This way, MLflow will copy the entire ``src/`` directory under ``code/`` and you
 
     By the same reason, the ``code_paths`` option doesn't handle the relative import of ``code_paths=["../src"]``.
 
+Limitation of ``code_paths`` in loading multiple models with the same module name but different implementations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The current implementation of the ``code_paths`` option has a limitation that it doesn't support reloading the model with the same module name but different implementations. The following example illustrates this limitation.
+
+.. code-block:: python
+
+    import importlib
+    import shutil
+    import sys
+    import tempfile
+    from pathlib import Path
+
+    import mlflow
+
+    with tempfile.TemporaryDirectory() as tmpdir:
+        tmpdir = Path(tmpdir)
+        my_model_path = tmpdir / "my_model.py"
+        code_template = """
+    import mlflow
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input):
+            return [{n}] * len(model_input)
+    """
+
+        my_model_path.write_text(code_template.format(n=1))
+
+        sys.path.insert(0, str(tmpdir))
+        import my_model
+
+        # model 1
+        model1 = my_model.MyModel()
+        assert model1.predict(context=None, model_input=[0]) == [1]
+
+        with mlflow.start_run():
+            info1 = mlflow.pyfunc.log_model(
+                artifact_path="model",
+                python_model=model1,
+                code_paths=[my_model_path],
+            )
+
+        # model 2
+        my_model_path.write_text(code_template.format(n=2))
+        importlib.reload(my_model)
+        model2 = my_model.MyModel()
+        assert model2.predict(context=None, model_input=[0]) == [2]
+
+        with mlflow.start_run():
+            info2 = mlflow.pyfunc.log_model(
+                artifact_path="model",
+                python_model=model2,
+                code_paths=[my_model_path],
+            )
+
+        shutil.rmtree(tmpdir)
+        sys.path.remove(str(tmpdir))
+        sys.modules.pop("my_model")
```

Comment:
Can we add a comment for what these lines for?

Reply:
added!